src/foundry/common/grid/base.d.mts

```diff
@@ -0,0 +1,614 @@
+
+  /**
+   * Get the point translated in a direction by a distance.
+   * @param point     - The point that is to be translated.
+   * @param direction - The angle of direction in degrees.
+   * @param distance  - The distance in grid units.
+   * @returns The translated point.
+   */
+  abstract getTranslatedPoint(point: Point, direction: number, distance: number): Point;
+
+  /**
+   * Get the circle polygon given the radius in grid units for this grid.
+   * The points of the polygon are returned ordered in positive orientation.
+   * In gridless grids an approximation of the true circle with a deviation of less than 0.25 pixels is returned.
+   * @param radius - The radius in grid units.
+   * @param center - The center point of the circle.
+   * @returns The points of the circle polygon.
+   */
+  abstract getCircle(center: Point, radius: number): Point[];
+
+  /**
+   * Get the cone polygon given the radius in grid units and the angle in degrees for this grid.
+   * The points of the polygon are returned ordered in positive orientation.
+   * In gridless grids an approximation of the true cone with a deviation of less than 0.25 pixels is returned.
+   * @param origin    - The origin point of the cone.
+   * @param radius    - The radius in grid units.
+   * @param direction - The direction in degrees.
+   * @param angle     - The angle in degrees.
+   * @returns The points of the cone polygon.
+   */
+  getCone(origin: Point, radius: number, direction: number, angle: number): Point[];
+
+  /* -------------------------------------------- */
+  /*  Deprecations and Compatibility              */
+  /* -------------------------------------------- */
+
+  /**
+   * Determine a placeable's bounding box based on the size of the grid.
+   * @param w - The width in grid spaces.
+   * @param h - The height in grid spaces.
+   * @deprecated Since v12 until v14. If you need the size of a Token, use Token#getSize instead.
+   */
+  getRect(w: number, h: number): PIXI.Rectangle;
+
+  /**
+   * Calculate the total size of the canvas with padding applied, as well as the top-left co-ordinates of the inner
+   * rectangle that houses the scene.
+   * @param gridType - The grid type to calculate padding for.
+   * @param width    - The width of the scene.
+   * @param height   - The height of the scene.
+   * @param size     - The grid size.
+   * @param padding  - The percentage of padding.
+   * @param options  - Options to configure the padding calculation.
+   * @deprecated Since v12 until v14. Use {@link BaseGrid#calculateDimensions} instead.
+   */
+  static calculatePadding(
+    gridType: foundry.CONST.GRID_TYPES,
+    width: number,
+    height: number,
+    size: number,
+    padding: number,
+    options?: InexactPartial<{
+      /**
+       * Are we computing padding for a legacy scene?
+       */
+      legacy: boolean;
+    }>,
+  ): { width: number; height: number; x: number; y: number };
```

Comment:
Similar factor out consideration, although less pressing due to deprecation

Reply:
I don't think this is worth a helper type. People shouldn't be using it anyway. Might push someone to update their code 😄